New version: QuantumCircuitOpt v0.6.0

JuliaRegistrator opened the pull request:
- Registering package: QuantumCircuitOpt
- Repository: https://github.com/harshangrjn/QuantumCircuitOpt.jl
- Created by: @harshangrjn
- Version: v0.6.0
- Commit: 350fb9a9161b48e93eab849600a06447937e5797
- Reviewed by: @harshangrjn
- Reference: https://github.com/harshangrjn/QuantumCircuitOpt.jl/issues/60#issuecomment-2584740121
- Description: A Julia/JuMP Package for Optimal Quantum Circuit Design
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
## Breaking changes

- Added support for `unitary_complex_conjugate` constraints (for both with and without global phase), which significantly speeds up QCOpt’s decomposition methods.
- Added a generic `multi_controlled_gate` function to evaluate multi-qubit gates with multiple controls for any single-qubit target gate (e.g. multiple-control Toffoli)
- Added 2-qubit Fermionic SWAP gate, that swaps adjacent fermionic modes
- Added `QFT3Gate` to docs
- Dropped support for `controlled_gate`
- Cleaned up angle discretization code, which is generic now - `get_all_angle_gates` and `get_discretized_angle_gates` 
- Qubit routing example (4-qubit gate) added

[noblock]
`````
<!-- END RELEASE NOTES -->

<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a83598d36fce98be54d0f538cc025bee2a4691220b7b3b2ba1740adfcced204477a77062cb1d4d6042451e2cd543ef732935624d5049f84452c1ea56bc43ddf67e4b7c5753e3f0aefa12d6aca44b0faf252c32af5c946385e678c87b60dc0432cf339a1a5429f2381685e163ddf0e4fcc531d7a48af17df73ba24a94f9235e74a89314 -->

harshangrjn commented:
Release notes:

## Breaking changes

- Added support for `unitary_complex_conjugate` constraints (for both with and without global phase), which significantly speeds up QCOpt’s decomposition methods.
- Added a generic `multi_controlled_gate` function to evaluate multi-qubit gates with multiple controls for any single-qubit target gate (e.g. multiple-control Toffoli)
- Added 2-qubit Fermionic SWAP gate, that swaps adjacent fermionic modes
- Added `QFT3Gate` to docs
- Dropped support for `controlled_gate`
- Cleaned up angle discretization code, which is generic now - `get_all_angle_gates` and `get_discretized_angle_gates` 
- Qubit routing example (4-qubit gate) added

harshangrjn commented:
I added the release notes in the original comment, and here too. Let me know if I am missing anything to release v0.6.0

harshangrjn commented:
Could you let me know why the automatic merging is still blocked?

IanButterworth commented:
Any comment you leave that isn't marked with [noblock] will block automerge.

https://github.com/JuliaRegistries/General?tab=readme-ov-file#automerge-is-blocked-by-one-of-my-comments-how-do-i-unblock-it

[noblock]

JuliaRegistrator committed:
New version: QuantumCircuitOpt v0.6.0

UUID: 88128e30-b60a-4e54-ab02-1050a5f92a36
Repo: https://github.com/harshangrjn/QuantumCircuitOpt.jl.git
Tree: 1e5bc0663b764726699f403bc0a2d3ed1f33920f

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
